libqtile/backend/wayland/qw/output.c

```diff
@@ -156,3 +156,159 @@ void qw_server_output_new(struct qw_server *server, struct wlr_output *wlr_outpu
         wlr_output_layout_add_auto(server->output_layout, wlr_output);
     wlr_scene_output_layout_add_output(server->scene_layout, l_output, output->scene);
 }
+
+void qw_output_background_destroy(struct qw_output *output) {
+    // Destroy the color scene_rect if it exists
+    struct wlr_scene_rect *color_rect = output->background.color_rect;
+    if (color_rect) {
```

Comment:
in discord we decided it might be better to do more explicit NULL checks

!= NULL